bin/deliver_reviewed_data.py

```diff
@@ -148,83 +214,112 @@ def get_list_of_file_to_move(self, sample_name):
+                working_dir=self.staging_dir,
+                cpus=1,
+                mem=2,
+                log_commands=False
+            )
+
+    def cleanup(self):
+        if self.no_cleanup:
+            return
+        if os.path.exists(self.staging_dir):
+            shutil.rmtree(self.staging_dir)
+
+
     def deliver_data(self, project_id=None, sample_id=None):
-        delivery_source = cfg.query('delivery_source')
         delivery_dest = cfg.query('delivery_dest')
         project_to_samples = self.get_deliverable_projects_samples(project_id, sample_id)
-        all_samples = []
         project_to_delivery_folder = {}
-
+        sample2stagedirectory={}
         for project in project_to_samples:
-            samples = project_to_samples.get(project)
-            self.info('deliver %s samples from %s'%(len(samples), project))
-            batch_delivery_folder = self.append_create_batch_delivery_folder(delivery_dest, project)
-            project_to_delivery_folder[project] = batch_delivery_folder
-            for sample in samples:
-                origin_sample_dir = os.path.join(delivery_source, project, sample)
-                if not os.path.isdir(origin_sample_dir):
-                    raise DataDeliveryError('Directory for sample %s in project %s does not exist'%(sample, project))
-                sample_folder = self.append_create_sample_folder(batch_delivery_folder, sample)
-                list_of_file_to_move = self.get_list_of_file_to_move(sample_name=sample)
-
-                for file_to_move in list_of_file_to_move:
-                    origin_file = os.path.join(origin_sample_dir, file_to_move)
-                    if not os.path.isfile(origin_file):
-                        raise DataDeliveryError('File %s for sample %s in project %s does not exist'%(file_to_move, sample, project))
-                    self.append_move_file_to_sample_folder(origin_file, sample_folder)
-                all_samples.append(sample)
+            for sample in project_to_samples.get(project):
+                stage_directory = self.stage_data(sample)
+                sample2stagedirectory[sample.get(ELEMENT_SAMPLE_INTERNAL_ID)] = stage_directory
 
         if self.dry_run:
-            print('====== Commands ======')
-            print('\n'.join(self.all_commands))
-            print('====== Metrics ======')
-            for project in project_to_delivery_folder:
-                header, lines = self.summarise_metrics_per_sample(project, project_to_delivery_folder.get(project))
+            print('Will Execute ')
+            print('\n'.join(self.all_commands_for_cluster))
+            print('Will move')
+            for project in project_to_samples:
+                today = datetime.date.today().isoformat()
+                batch_delivery_folder = os.path.join(delivery_dest, project, today)
+                print('%s --> %s'%(sample2stagedirectory.get(sample.get(ELEMENT_SAMPLE_INTERNAL_ID)), batch_delivery_folder))
```

Comment:
I'm getting a 'might be referenced before assignment' warning on `sample` at line 296. Is this the intended behaviour?